File: server/src/models/certification.rs
use serde::Serialize;
use sqlx::FromRow;

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct CertificationModel {
    pub id: i32,
    pub user_id: i32,
    pub name: String,
    pub organization: String,
    pub issue_date: Option<String>,
    pub expiration_date: Option<String>,
    pub credential_id: Option<String>,
    pub credential_url: Option<String>,
    pub created_at: String,
}

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct PublicCertificationModel {
    pub name: String,
    pub organization: String,
    pub issue_date: Option<String>,
    pub expiration_date: Option<String>,
    pub credential_id: Option<String>,
    pub credential_url: Option<String>,
}

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct AuthCertificationModel {
    pub id: i32,
    pub name: String,
    pub organization: String,
    pub issue_date: Option<String>,
    pub expiration_date: Option<String>,
    pub credential_id: Option<String>,
    pub credential_url: Option<String>,
}

File: src/types/certification.ts
export interface CertificationModel {
  id: number;
  user_id: number;
  name: string;
  organization: string;
  issue_date?: string;
  expiration_date?: string;
  credential_id?: string;
  credential_url?: string;
  created_at: string;
}

export interface PublicCertificationModel {
  name: string;
  organization: string;
  issue_date?: string;
  expiration_date?: string;
  credential_id?: string;
  credential_url?: string;
}

export interface AuthCertificationModel extends PublicCertificationModel {
  id?: number;
}

export interface AddCertificationPayload {
  name: string;
  organization: string;
  issueDate?: string;
  expiration_date?: string;
  credential_id?: string;
  credential_url?: string;
}
